client/src/shared/PlayerListSort.tsx

```diff
@@ -0,0 +1,101 @@
+import { Button } from "@radix-ui/themes";
+import { MdSort } from "react-icons/md";
+import React from "react";
+import { useDefiniteGame } from "../store/GameContext";
+import { getCharacter } from "../assets/game_data/gameData";
+
+interface PlayerListSortProps<T extends string> {
+  allSorts: Record<T, unknown[]>;
+  selectedSort?: T;
+  setSelectedSort: (sort: T) => void;
+}
+
+export function PlayerListSort<T extends string>({
+  allSorts,
+  selectedSort = Object.keys(allSorts)[0] as T,
+  setSelectedSort,
+}: PlayerListSortProps<T>) {
+  const keyRing = React.useMemo(
+    () =>
+      Object.keys(allSorts).reduce(
```

Comment:
why are you sorting in all ways each time?  You don't need counts here like you did for filters.